### ISSUE TITLE
region using aria-labelledby doesn't have name of region read in all circumstances

### ISSUE THREAD

**Issue by Epic-Third-Party-Bug-Reporting** [nvaccess/nvda#19094]
### Steps to reproduce

1.	Find a region which derives its accessible name from aria-labelledby and  which contains more content than just the element aria-labelledby points at.
(e.g. [the "Second" region in this codepen](https://codepen.io/Kaulabau/details/myVeWNR))
2.	Turn on NVDA and position the virtual cursor outside of the region
3.	Move the virtual cursor into the region using the arrow keys such that the navigator object does not land on the element aria-labelledby points at

### Actual behavior

NVDA reports that you have entered a region, but does not report its name.

### Expected behavior

NVDA should include the region's name.

### NVDA type

installed copy

### NVDA version

2025.2.0.37684, 2025.3.0.52596

### Windows version

Windows 11 Enterprise 24H2 OS build 26100.6584

### Name and version of other software in use when reproducing the issue

Chrome version 140.0.7339.186 (Official Build) (64-bit)

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I have restarted NVDA with add-ons disabled and the issue still occurs

### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

I have run the COM Registration Fixing Tool and the issue still occurs

**Comment by josephsl** [comment]
Hi,

Thanks. Three follow-up questions:

1. Is this happening across screen readers .eg. Narrator and NVDA as opposed to just NVDA?
2. Can you report back after testing the scenario with Firefox and other web browsers?
3. Are there additional example websites/documents exhibiting this issue?

Terminology note: we (NVDA community) use "browse mode" instead of virtual cursor.

Thanks.

**Comment by Epic-Third-Party-Bug-Reporting** [comment]
Sure, sorry for forgetting to add those originally! The described behavior also occurs in Firefox (version 144.0) and Edge (version 141.0.3537.92). JAWS 2025 reads region names as expected in all three browsers listed. Narrator does as well. I haven't found many websites in the wild that are using aria-labelledby instead of aria-label on their sections/regions.
And thanks for the terminology note!
